engine/classes/Elgg/Application.php

```diff
@@ -0,0 +1,293 @@
+<?php
+
+namespace Elgg;
+
+/**
+ * WARNING: API IN FLUX. DO NOT USE DIRECTLY.
+ *
+ * @access private
+ *
+ * @package    Elgg.Core
+ * @subpackage Application
+ * @since      2.0.0
+ */
+class Application {
+
+	/**
+	 * @var \Elgg\Config
+	 */
+	private $config;
+
+	/**
+	 * @var string
+	 */
+	private $engineDir;
+
+	/**
+	 * @var string
+	 */
+	private $wwwRoot;
+
+	public function __construct() {
+		$this->config = new \Elgg\Config((object)array());
+
+		$this->engineDir = dirname(dirname(dirname(__FILE__)));
+		$this->wwwRoot = dirname($this->engineDir);
+	}
+
+	/**
+	 * Loads all lib files, but doesn't trigger boot events.
+	 */
+	function loadConfig() {
+		/**
+		 * The time with microseconds when the Elgg engine was started.
+		 *
+		 * @global float
+		 */
+		global $START_MICROTIME;
+		$START_MICROTIME = microtime(true);
+
+		/**
+		 * Configuration values.
+		 *
+		 * The $CONFIG global contains configuration values required
+		 * for running Elgg as defined in the settings.php file.
+		 *
+		 * Plugin authors are encouraged to use elgg_get_config() instead of accessing
+		 * the global directly.
+		 *
+		 * @see elgg_get_config()
+		 * @see engine/settings.php
+		 * @global \stdClass $CONFIG
+		 */
+		global $CONFIG;
+		if (!isset($CONFIG)) {
+			$CONFIG = new \stdClass;
+		}
+		$CONFIG->boot_complete = false;
+
+
+		// No settings means a fresh install
+		if (!is_file("{$this->engineDir}/settings.php")) {
```

Comment:
FYI it's just the existing code moved around. It certainly could use improvements, will have a look at it.